function "error_func" is already defined
